crates/sqllib/src/binary.rs

```diff
@@ -60,6 +64,23 @@ impl ByteArray {
         Self { data: d.to_vec() }
     }
 
+    pub fn with_size(d: &[u8], size: i32) -> Self {
+        if size < 0 {
+            ByteArray::new(d)
+        } else {
+            let size = size as usize;
+            match d.len().cmp(&size) {
+                Ordering::Equal => ByteArray::new(d),
+                Ordering::Greater => ByteArray::new(&d[..size]),
+                Ordering::Less => {
+                    let mut data: Vec<u8> = vec![0; size];
+                    data[..d.len()].copy_from_slice(d);
```

Comment:
if you add a `ByteArray::from_vec` function that passes `data` directly you can save an additional allocation here